src/pages/workspaces/workspace/_tests/WorkspaceMenu.test.js

```diff
@@ -0,0 +1,258 @@
+  })
+})
+
+
+describe('WorkspaceMenu - GCP workspace', () => {
+  beforeEach(() => {
+    useWorkspaceDetails.mockReturnValue({ workspace: { canShare: false, workspace: { isLocked: false } } })
+    MenuTrigger.mockImplementation(({ content }) => { return div([content]) })
+    TooltipTrigger.mockImplementation(({ content, children }) => {
+      const [open, setOpen] = useState(false)
+      return (div([
+        div(
+          {
+            onMouseEnter: () => {
+              setOpen(true)
+            },
+            onMouseLeave: () => {
+              setOpen(false)
+            }
+          },
+          [children]
+        ),
+        open && !!content && div([content])
+      ]))
+    })
+  })
+
+  it('renders Clone menu item as disabled', () => {
+    const onClone = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onClone },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    const menuText = screen.getByText('Clone')
+    expect(menuText).not.toHaveAttribute('disabled')
+  })
+
+  it('renders Clone tooltip', () => {
+    const onClone = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onClone },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    fireEvent.mouseOver(screen.getByText('Clone'))
+    expect(screen.queryByText('Cloning is not currently supported on Azure Workspaces')).toBeNull()
+  })
+
+  it('renders Share menu item as disabled', () => {
+    const onShare = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onShare },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    const menuText = screen.getByText('Share')
+    expect(menuText).toHaveAttribute('disabled')
+  })
+
+  it('renders Share menu item as enabled', () => {
+    useWorkspaceDetails.mockReturnValue({ workspace: { canShare: true, workspace: { isLocked: false } } })
+    const onShare = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onShare },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    const menuText = screen.getByText('Share')
+    expect(menuText).not.toHaveAttribute('disabled')
+  })
+
+  it('renders Share tooltip', () => {
+    const onShare = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
+      callbacks: { onShare },
+      workspaceInfo: { name: 'example1', namespace: 'example-billing-project' }
+    }))
+    fireEvent.mouseOver(screen.getByText('Share'))
+    screen.getByText('You have not been granted permission to share this workspace')
+    expect(screen.queryByText('Sharing is not currently supported on Azure Workspaces')).toBeNull()
```

Comment:
Let's not test the tooltip text that shouldn't be present given that you are testing what actually is present.